src/main/java/org/candlepin/service/impl/DefaultUserServiceAdapter.java

```diff
@@ -108,7 +107,7 @@ public UserInfo createUser(UserInfo user) {
         User entity = new User();
 
         entity.setUsername(user.getUsername());
-        entity.setHashedPassword(user.getHashedPassword());
+        entity.setPassword(user.getPassword());
```

Comment:
This is (one of) the point(s) at which we should be doing whatever hashing we're going to be doing. Previously we were relying on the password already being hashed when we got to this point, but now we're just setting everything in clear text -- we're doing no hashing at all:

```
candlepin=> SELECT * FROM cp_user;
                id                |         created         |         updated         | hashedpassword | superadmin | username  
----------------------------------+-------------------------+-------------------------+----------------+------------+-----------
 ff8081818d7fad30018d7fad4b4c0001 | 2024-02-06 18:27:59.948 | 2024-02-06 18:27:59.948 | admin          | t          | admin
 ff8081818d7fad30018d7fad56330005 | 2024-02-06 18:28:02.739 | 2024-02-06 18:28:02.739 | password       | f          | testuser1
 ff8081818d7fad30018d7fad56540006 | 2024-02-06 18:28:02.772 | 2024-02-06 18:28:02.772 | password       | f          | testuser2
 ff8081818d7fad30018d7fad56700007 | 2024-02-06 18:28:02.8   | 2024-02-06 18:28:02.8   | password       | f          | testuser3
 ff8081818d7fad30018d7fad56880008 | 2024-02-06 18:28:02.824 | 2024-02-06 18:28:02.824 | password       | t          | doc
 ff8081818d7fad30018d7fad56a20009 | 2024-02-06 18:28:02.85  | 2024-02-06 18:28:02.85  | password       | f          | grumpy
 ff8081818d7fad30018d7fad56ba000a | 2024-02-06 18:28:02.874 | 2024-02-06 18:28:02.874 | password       | f          | happy
 ff8081818d7fad30018d7fad56d1000b | 2024-02-06 18:28:02.897 | 2024-02-06 18:28:02.897 | password       | f          | sleepy
 ff8081818d7fad30018d7fad56e9000c | 2024-02-06 18:28:02.921 | 2024-02-06 18:28:02.921 | password       | f          | bashful
 ff8081818d7fad30018d7fad5701000d | 2024-02-06 18:28:02.945 | 2024-02-06 18:28:02.945 | password       | f          | sneezy
 ff8081818d7fad30018d7fad571a000e | 2024-02-06 18:28:02.97  | 2024-02-06 18:28:02.97  | password       | f          | dopey
 ff8081818d7fad30018d7fad5731000f | 2024-02-06 18:28:02.993 | 2024-02-06 18:28:02.993 | password       | t          | huey
 ff8081818d7fad30018d7fad57480010 | 2024-02-06 18:28:03.016 | 2024-02-06 18:28:03.016 | password       | f          | duey
 ff8081818d7fad30018d7fad575d0011 | 2024-02-06 18:28:03.037 | 2024-02-06 18:28:03.037 | password       | f          | louie
 ff8081818d7fad30018d7fad57740012 | 2024-02-06 18:28:03.06  | 2024-02-06 18:28:03.06  | password       | f          | mickey
 ff8081818d7fad30018d7fad57a80013 | 2024-02-06 18:28:03.112 | 2024-02-06 18:28:03.112 | password       | f          | minnie
 ff8081818d7fad30018d7fad57bf0014 | 2024-02-06 18:28:03.135 | 2024-02-06 18:28:03.135 | password       | f          | magoo
```

Since the UserServiceAdapter implementation is responsible for both persisting and validating user credentials, it can do whatever it wants to make the comparison without the hashing being exposed at the generic Util level, or affecting other adapter logic in any way. We could move the bcrypt logic over to this class with a util function as follows:

```
private String hashPassword(*byte[] salt,* String password) {
  // bcrypt stuff here
  // salt param is optional at this layer since it's all internal anyway, but it 
  // does make it abundantly clear that a salt is in use and should be considered
  // if future changes are to be made.
}

...

byte[]|String salt = this.generateSalt(...); // could do all sorts of fun stuff here
entity.setPassword(this.hashPassword(salt, user.getPassword()));
```

Reply:
It, should be moved. Should we create card for the password handling what you mentioned above.